### ISSUE TITLE
G920 Problems

### ISSUE THREAD

**Issue by deviljin112** [berarma/oversteer#67]
Hello,

I seem to face a couple problems on my G920.
Firstly, the "tests" does not work properly, after the 2 initial tests, the third one (Steps measure) fails with "Steering wheel not responding." error.
Secondly, there seems to be a big problem with inverted pedals, as G920 works as inverted by default, some games (like F1 2020) makes the brake pedal (in my case) always be at 100%. Not sure if this is a game issue or if there's an option in the driver to flip it around as the game does not allow me to change the inverted status. Further, when remapped to any other pedal the brake simply doesn't work (although i can see it work on oversteer). Similar inverted behavior can be found in other games, but most have an option to "invert" the pedal which resolves the issue.

Any help would be appreciated.
Thank you

Edit: I do realize the second issue could be related to Proton as well, or F1 being wonky however, from protondb it would appear that most people have no issues with both G920 and G29. Maybe new update (F1) broke something? Really confused about this.

**Comment by berarma** [comment]
The only thing in this message that is related to Oversteer is the error in the FFB tests. I can take a look and maybe ask for some more info.

I know from other people that F1 2020 on Proton is very tricky when it comes to steering wheels (and pedals). I can't help you. You should look for the issue in the Proton repository.

**Comment by deviljin112** [comment]
Yeah I figured (about second problem) was worth asking anyway. Happy to assist with anything regarding the first issue, the wheel begins the test spins about 90* to the right and the error appears. I tried running with `--debug` but that didn't flag any problems.

**Comment by deviljin112** [comment]
Okey so I somehow fixed my 2nd issue, see [here](https://github.com/ValveSoftware/Proton/issues/4047#issuecomment-868995335)
tldr;

- Before going into Steering settings Hold down Brake Pedal
- Let go of the brake once the settings appear
- Remap brake to the brake pedal (will appear as "Z-")
- Save the configuration
- Re-enter the configuration and now when you press brake it will respond as expected

But the FFB is very strange, it feels like the wheel is applying FFB without even the game running and constantly spinning both left and right (pulsing lightly). Doesn't occur without Oversteer, but without oversteer there's no FFB at all and also can't limit my range to 270*. Any ideas what it may be?? Could the "test" issue impact how the FFB is applied to the wheel?

**Comment by berarma** [comment]
If the test feature fails it might have an effect, maybe, but I think it's improbable.

Anyway, please stop trying to relate the F1 issues to Oversteer. This is a passive application and has nothing to do with the way games work. You can easily try it without Oversteer and use what's best for you.

If I could help I'd try but I don't even have that game. You'll.get more help in the Proton issue.

**Comment by isopix** [comment]
Maybe the problem is that Oversteer provides in your case udev rules (you can easily check that trough package manager), that gives you 'rw' access to the wheel, and without it, F1 cannot 'wire' to device (send force feedback events)

It's only guess.
You can also check if Oversteer is reset to default.
If you would use G29(or older wheel) and new-lg4ff driver, than you could blame Bernat. But for G920 it's not the case unfortunately ;-)

**Comment by deviljin112** [comment]
> If the test feature fails it might have an effect, maybe, but I think it's improbable.
> 
> Anyway, please stop trying to relate the F1 issues to Oversteer. This is a passive application and has nothing to do with the way games work. You can easily try it without Oversteer and use what's best for you.
> 
> If I could help I'd try but I don't even have that game. You'll get more help in the Proton issue.

My intention wasn't to relate the F1 to Oversteer but rather make it easier to find help regarding G920 wheel, as I found the above solution to also be applicable to other games which dont allow for manually setting inverted axis. I do apologize I wasn't aware that you would rather not have those related. It's just really hard to find any help/support regarding the G920 wheel on linux from what I've found.

Running F1 without oversteer means no FFB at all (not sure why). I found a temporary solution of reducing FFB on oversteer to 50% but it still has strange behaviors occasionally where the wheel would be pushing heavier to one side than the other and switching sides after few minutes. I'm not blaming Bernat for the fault, I'm simply seeing if there's a simple solution to this, my intentions where not ill. I am aware that Bernat does not have a G920 and debugging it would be difficult. As i said above F1 issue has been resolved and as mentioned unrelated to Oversteer I just posted the solution in case another person finds this post. I'm just curious about the issue with FFB test failure and whether there's any ideas for how to solve the strange FFB behavior. Thank you and I apologize for linking the issues and if my ticket has been taken negatively.

**Comment by berarma** [comment]
> > If the test feature fails it might have an effect, maybe, but I think it's improbable.
> > 
> > Anyway, please stop trying to relate the F1 issues to Oversteer. This is a passive application and has nothing to do with the way games work. You can easily try it without Oversteer and use what's best for you.
> > 
> > If I could help I'd try but I don't even have that game. You'll get more help in the Proton issue.
> 
> My intention wasn't to relate the F1 to Oversteer but rather make it easier to find help regarding G920 wheel, as I found the above solution to also be applicable to other games which dont allow for manually setting inverted axis. I do apologize I wasn't aware that you would rather not have those related. It's just really hard to find any help/support regarding the G920 wheel on linux from what I've found.
> 
> Running F1 without oversteer means no FFB at all (not sure why). I found a temporary solution of reducing FFB on oversteer to 50% but it still has strange behaviors occasionally where the wheel would be pushing heavier to one side than the other and switching sides after few minutes. I'm not blaming Bernat for the fault, I'm simply seeing if there's a simple solution to this, my intentions where not ill. I am aware that Bernat does not have a G920 and debugging it would be difficult. As i said above F1 issue has been resolved and as mentioned unrelated to Oversteer I just posted the solution in case another person finds this post. I'm just curious about the issue with FFB test failure and whether there's any ideas for how to solve the strange FFB behavior. Thank you and I apologize for linking the issues and if my ticket has been taken negatively. 

I'm just trying to be clear that F1 doesn't need Oversteer to work, or at least it shouldn't. You can remove Oversteer from your system and you should be able to play F1 just the same.

If that's not the case, that doesn't mean I'm able to fix F1. It's just some side-effect that I can't understand, but most probably the answer resides more in your system than in Oversteer.

I think you should find out why you can't use the wheel on F1 without Oversteer. What do you do on Oversteer so that F1 works? Apart from setting the range and the gain there's little more that affects the wheel.

**Comment by deviljin112** [comment]
Yes, I do understand that Oversteer is just passive and games dont need it to work. Also because G920 does have kernel support out the box.
I'm still trying to find the root cause of this, but to my understanding after reboot G920 runs "default" i.e. without any modifications, when in that mode, F1 (or any other game from codemasters it would appear - no issues with Assetto) simply has no FFB. When I turn on oversteer and reboot the game, FFB is back and works however it then proceeds to have this strange one side heavy FFB that moves from one side to the other every now and again. The problem I'm having is that while yes I know oversteer has no direct impact on the game, playing without it means zero FFB, and playing with it means fighting the wheel.
You might be right and the issue may be with my system I will continue doing some digging and see if there's a solution.
As for what I do? Actually nothing, regardless of the profile, the wheel suddenly gains FFB when oversteer is on (just with codemasters games). It may be the issue of how codemasters handle wheel input? I'm honestly not entirely sure but I am getting closer to isolating the root cause, as at first I thought the issue is persistent across other games but then realised I've tested it on codemasters games only. I might prune all of oversteer as well as the rules it created and see what behavior i get then. I dont know if you would like me to keep you updated on the process. I do realise that this issue while i began thinking was somewhat related to oversteer now appears to be outside of the scope.  Thanks

**Comment by berarma** [comment]
If you figure out why Oversteer seems to enable FFB for this game then I'd like to know. But there's nothing that I know that could influence it.

**Comment by isopix** [comment]
If you want possible help, then you should provide bug report, or at least:

1) Output of following command, before running Oversteer:
 'ls -lR /dev/input/'

2) Output of following command, before running Oversteer:
 'ls -lR /dev/input/'

3) Oversteer configuration file

4) If you installed Oversteer from package, like RPM, DEB, pkg.zst,
then you must attach this file (or provide link to it)

**Comment by deviljin112** [comment]
The above `ls` commands give the same output.
Configuration I'm running is default.
Changing the range doesn't affect the FFB behaviour.
I'm running the newest release of Oversteer, built from source.
And can safely say Oversteer has nothing to do with this issue.
After uninstalling and pruning all files, the same FFB behavior can be seen despite previously (strangely) relying on Oversteer to be turned on.
I've rebuilt it from source and ruled out Oversteer being at fault.

But, I may have found a more information (still figuring out the solution).
When running `journalctl -f` and playing F1 (just as an example) I get the following message:

`kernel: logitech-hidpp-device 0003:046D:C262.0007: Force feedback command queue contains 220 commands, causing substantial delays!`

So in fact the strange FFB isn't the wheel being weird, but the FFB is just really delayed hence why it behaves so strangely.
That same message can be found in `dmesg`.

Since this is definitely not an oversteer issue you are welcome to close the ticket and I apologize for wasting your time. If you however can / have any suggestions on how to resolve the above it would be greatly appreciated.

**Comment by isopix** [comment]
This kernel message is substantial, because new-lg4ff got supression
of to many command, which in-kernel driver lacks.

So you should open ticket in the kernel maybe.
but if Oversteer changes something, than it's strange, so I think it
can stay theere.

It sounds like fix should be trivial(not for too many commands)

But I have feeling that you are over-confident and you are 'lying'
about permissions in /dev/input. Have you used 'diff' command to
compare output
Simple attach command outputs if you want any help, because we
shouldn't rely on your feeling it's the same, but base our support on
hard facts.
Maybe you have right, and it's the same, but if not we can spend much
time trying to help what's wrong, and in the end we will find that you
missed one difference in permissions listing that caused problems. You
will feel stupid, and we will angry that we wasted out time because
your laziness/over-confidence.

So please post these damn command outputs and Oversteer configuration
file, because it's 99% probability that's udev rules, or you are
switching wheel in some compatiblity mode(with older wheels).
You should also point us to Oversteer package you are installing.

While even I can help you with udev rules and maybe some packaging
issues, I guess that Bernat can quickly look at your config file.

We even don't know which 'Oversteer' version you are using. Distro
packages can have broken udev rules. you can package everything from
/etc/udev/rules and /usr/share/udev/rules or /usr/lib/udev//rules.
There can be some stupid rules overrides provided by some package with
rules (especially joystick emulators, etc. I had some prob;ems in the
past with it and Razer Hydra). Maybe even you have copy-pasted some
stupid/buggy rules from internet to system and you even don't remember
you.

In the and I know you are using some F1 game, but there are many. And
you are using proton.
Please test with 'fftest' or some native games (You can buy F1 2015 or
F1 2017 for $4 when on promotion [or 3rd party reseller. Or some other
Codemasters native games like Dirt Rally 1. There were some other
native, but not sure [you must check it yourself, but probably some
Grid Autosport and some older 'Dirts' were native)

Personally I think that ticket is important, because there will be
others with similar problems and will file similar bugs. and fix is
probably trivial (for ffb not working, because it works for 99% of
other g920 users)

**Comment by berarma** [comment]
> Edit: I've just seen that ffbwrap has a throttling option which has solved issues with AC:C and G920. However after adding:
> 
> `/home/dev/Downloads/ffbtools-master/bin/ffbwrap --update-fix --throttling /dev/input/by-id/usb-Logitech_G920_Driving_Force_Racing_Wheel_for_Xbox_One_00005408d2a4768f-event-joystick -- %command%`
> 
> to steam launch options, the issue persists.

Have you followed the [build instructions](https://github.com/berarma/ffbtools/blob/master/docs/README.md#build)?

Have you seen any errors when compiling?

Issues with ffbtools can be forwarded to that project.

**Comment by berarma** [comment]
I don't think there could be anything wrong in the device permissions or Oversteer configuration that would produce the queue error and broken FFB. For me, the only issue seems to be that ffbwrap isn't working.

> As for building ffbwrap I have built it correctly with zero errors on compilation. I may have it in the wrong path but I'm not sure what the correct location or pathing should be for this.
> Please let me know if you need anything else.

You can download and build ffbtools anywhere as long as you run the script in-place. It needs to find the libraries in the build directory.

You can try running ffbwrap from the terminal. Any command will do.

I'm closing this issue since most of the talk has nothing to do with Oversteer.

Please, open an issue for ffbtools pasting the output of the build command and a ffbwrap test running any command from a terminal.

Please, open another issue for the FFB test error in case you want to help me fix it.

Thanks.
